mvc-api/mvc-service/app/managers/CSVContentProcessor.java

```diff
@@ -0,0 +1,98 @@
+package managers;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.sunbird.search.util.SearchConstants;
+
+import java.io.*;
+import java.util.Arrays;
+
+public class CSVContentProcessor {
+GetContentDefinition getContentDefinition = new GetContentDefinition();
+    JSONObject dataKeys;
+    JSONParser parser = new JSONParser();
+    public void readCsvFile(File csvFile) {
+        String[] header = {};
+        int rowCount = 0;
+
+        try {
+            dataKeys = (JSONObject) parser.parse(SearchConstants.contentKeysObj);
+            try (BufferedReader br = new BufferedReader(new FileReader(csvFile))) {
+                String line;
+
+                while ((line = br.readLine()) != null) {
+                    if (rowCount == 0) {
+                        header = line.split(SearchConstants.csvfiledelimeter);
+                    } else {
+                        String[] values = line.split(SearchConstants.csvfiledelimeter);
+                        JSONObject data = getJsonObjectForValues(header, values);
+                        // Do the processing
+                        // get source url
+                        String sourceurl = data.get("sourceURL").toString();
+                        sourceurl = sourceurl.replaceAll("[\\n\\t ]", "");
+                        if (sourceurl.contains("?mode=edit")) {
+                            sourceurl = sourceurl.toString().substring(0, sourceurl.toString().lastIndexOf('?'));
+                        }
+                        // get content definition
+                        getContentDefinition.getDefinition(data,sourceurl);
+                    }
+
+                    rowCount++;
+                }
+            } catch (FileNotFoundException e) {
+
+            } catch (IOException e) {
+
+            }
+        }
+        catch (Exception e) {
+
+        }
```

Comment:
Exceptions are suppressed. If the data is not valid then it won't process those records.

Reply:
will make the change